client/services/utils/localSettings.js

```diff
@@ -0,0 +1,143 @@
+const STORAGE_KEY = 'localsettings'
+const DATA_TYPE_KEY = 'dataType'
+const VALUE_KEY = 'value'
+class LocalSettings {
+  constructor ($window) {
+    Object.assign(this, { $window })
+    this.listeners = {}
+    this.STORAGE_KEY = STORAGE_KEY
+    const fromSession = this.get()
+    if (fromSession !== undefined) {
+      this.settingsObject = fromSession
+    } else {
+      this.settingsObject = { }
+    }
+    this.update(this.settingsObject, 'self', () => {})
+  }
+
+  parseSettingString (settingsToUpdate) {
+    const parsedSettings = {}
+    Object.keys(settingsToUpdate).forEach(key => {
+      if (settingsToUpdate[key].hasOwnProperty(DATA_TYPE_KEY) &&
+      settingsToUpdate[key].hasOwnProperty(VALUE_KEY)) {
+        const dataType = settingsToUpdate[key][DATA_TYPE_KEY]
+        // TODO: add the other datatypes
+        // TODO: make it not fail silently
+        switch (dataType.toLowerCase()) {
+          case 'string':
+            parsedSettings[key] = settingsToUpdate[key][VALUE_KEY]
+            break
+          case 'boolean':
+            parsedSettings[key] = (settingsToUpdate[key][VALUE_KEY] === 'true')
+            break
+        }
+      }
+    })
+    return parsedSettings
+  }
+
+  checkSettingValidity (setting) {
+    if (setting.hasOwnProperty('name') && setting.hasOwnProperty('dataType') && setting.hasOwnProperty('value')) {
+      const dataType = setting[DATA_TYPE_KEY]
+      let valid = false
+      if (dataType === 'number' || dataType === 'boolean' || dataType === 'string') {
+        // eslint-disable-next-line valid-typeof
+        valid = (typeof (setting[VALUE_KEY]) === dataType)
+      }
+      return valid
+    }
+    return false
+  }
+
+  addSettings (sourceName, settingsArray) {
```

Comment:
changed to `namespace`. it makes more sense. however, the use of `array` in the name means that the expected input is an array, rather than an object...